py/string.go

```diff
@@ -34,6 +34,22 @@ or repr(object).
 encoding defaults to sys.getdefaultencoding().
 errors defaults to 'strict'.`, StrNew, nil)
 
+
+func init() {
+	StringType.Dict["split"] = MustNewMethod("split", func(self Object, value Object) (Object, error) {
+		selfStr := self.(String)
+		if valStr, ok := value.(String); ok {
+			ss := strings.Split(string(selfStr), string(valStr))
+			o := List{}
+			for _, j := range ss {
+				o.Items = append(o.Items, String(j))
+			}
+			return &o, nil
+		}
+		return nil, fmt.Errorf("Not split by string")
+	}, 0, "split(sub) -> split string with sub.")
```

Comment:
The Python help is

```
    S.split([sep [,maxsplit]]) -> list of strings
    
    Return a list of the words in the string S, using sep as the
    delimiter string.  If maxsplit is given, at most maxsplit
    splits are done. If sep is not specified or is None, any
    whitespace string is a separator and empty strings are removed
    from the result.
```

Which makes me see that we are missing two things.
- If `sep` isn't passed (or is None) in then we should be using `strings.Fields` in go terms (This doesn't have a maxfields parameter though.)
- There is another optional parameter to specify the number of splits.

Here are some cases to consider

```
>>> "a,d,c".split(",")
['a', 'd', 'c']
>>> "a,d,c".split(",",1)
['a', 'd,c']
>>> " a   d   b   ".split()
['a', 'd', 'b']
>>> " a   d   b   ".split(None, 1)
['a', 'd   b   ']
```